services/query-cmr/app/query.ts

```diff
@@ -206,67 +163,99 @@ export async function validateGranules(
-    const { requestId, sources, unencryptedAccessToken } = operation;
-    logger.info(`Querying granules for ${sources[0].collection}`, { collection: sources[0].collection });
-    const startTime = new Date().getTime();
+    // Include OPeNDAP links in the response unless the data operation explicitly overrides
+    // by setting extraArgs.includeOpendapLinks to false
+    const includeOpendapLinks = operation.extraArgs?.include_opendap_links !== false;
+    [totalItemsSize, outputItemSizes, catalogs, newScrollId, hits] =
+      await thisModule.querySearchAfter(
+        requestId, unencryptedAccessToken, scrollId, maxCmrGranules, logger, includeOpendapLinks,
+      );
+  } catch (e) {
+    if (e instanceof RequestValidationError || e instanceof CmrError) {
+      if (granValidation) {
+        // Avoid giving confusing errors about GeoJSON due to upstream converted files
+        if (e.message.indexOf('GeoJSON') !== -1 && shapeType) {
+          e.message = e.message.replace('GeoJSON', `GeoJSON (converted from the provided ${shapeType})`);
+        }
+        return {
+          hits,
+          error: e.message,
+          errorLevel: 'error',
+          errorCategory: 'granValidation',
+        };
+      } else {
+        throw (e);
+      }
+    } else {
+      throw new ServerError('Failed to query the CMR');
+    }
+    logger.error(e);
+  }
 
-    const { hits } = await queryGranulesWithSearchAfter(
-      { 'id': requestId },
-      unencryptedAccessToken,
-      maxCmrGranules,
-      {},
-      scrollId,
-    );
+  if (granValidation) {
     if (hits === 0) {
       return {
-        hits: 0,
+        hits,
         error: 'No matching granules found.',
         errorLevel: 'error',
         errorCategory: 'granValidation',
       };
     }
-    jobHits = hits;
+
     const msTaken = new Date().getTime() - startTime;
     logger.info('timing.cmr-validate-granule.end', { durationMs: msTaken, hits });
 
     const limitedMessage = getResultsLimitedMessageImpl(hits, maxResults, maxCmrGranules, reason, serviceName, hasGranuleLimit, sources[0].collection);
     if (limitedMessage) {
       jobMessages.push(limitedMessage);
     }
-  } catch (e) {
-    if (e instanceof RequestValidationError || e instanceof CmrError) {
-      // Avoid giving confusing errors about GeoJSON due to upstream converted files
-      if (e.message.indexOf('GeoJSON') !== -1 && shapeType) {
-        e.message = e.message.replace('GeoJSON', `GeoJSON (converted from the provided ${shapeType})`);
-      }
-      logger.error(e);
-      return {
-        hits: 0,
-        error: e.message,
-        errorLevel: 'error',
+
+    // use errorCategory: 'granValidation' to indicate success with the job hits and message
+    if (jobMessages.length > 0) {
+      result = {
+        hits,
+        scrollID: scrollId,
+        error: jobMessages.join(' '),
+        errorLevel: 'warning',
+        errorCategory: 'granValidation',
+      };
+    } else {
+      result = {
```

Comment:
I'm not positive - but I think this might be the success scenario. In the success scenario we shouldn't be setting the errorLevel and errorCategory. I think this might be the cause of the status showing up as warning:granValidation in the workflow-ui when forceAsync is true even for successful granules.